hutch_python/utils.py

```diff
@@ -433,3 +436,22 @@ def get_fully_qualified_domain_name():
         )
         logger.debug("getfqdn failed", exc_info=True)
         return ""
+
+
+abort_msg = """
+Your RunEngine has been interrupted.  If you aborted the current run (Ctrl+C),
+the RunEngine is ready for use.  If you paused the run, you must resume,
+abort, stop, halt the RunEngine before continuing.
+"""
+
+
+class AbortSigintHandler(SignalHandler):
+    def __init__(self, RE):
+        super().__init__(signal.SIGINT, log=RE.log)
+        self.RE = RE
+
+    def handle_signals(self):
+        # Check for pause requests from keyboard.
+        if self.RE.state.is_running and (not self.RE._interrupted):
+            threading.Thread(target=self.RE.abort(reason='sigint')).start()
```

Comment:
I think it's worth pausing for a moment to make sure that `abort` is the one we want. After all, we could also have picked `stop` or `halt`.

Semantically, an abort is a failed run, but we don't necessarily know if the run was successful or failed just because it was interrupted early.

Practically, an abort raises an exception that creates a long traceback in the terminal, something a stop does not do.

Any thoughts about this?

Reply:
We can't guess the user's intentions, of course. In the interest of avoiding tracebacks which will be the subject of many complaints, `stop` seems appropriate.

Reply:
I guess I didn't balk at the ~8 lines of exception traceback, though getting rid of that is certainly nice.  

I generally wouldn't consider any interrupted run "successful", even if intentional.  Exit statuses aren't recorded anywhere by the daq, so the distinction between `stop` and `abort` might just be a superficial one.

`halt` doesn't do any cleanup, so I'd shy away from that option.

Reply:
I think the the thing that bugs me about the abort traceback is that it's pretty useless- is it ever different?
Agree on `halt` not being a good choice

Reply:
I can get onboard with `stop`.   The traceback from the `RunEngineInterrupted` exception tends to be descriptive enough.

Reply:
As Zach noted it's almost always the same - it's not as if we're catching KeyboardInterrupt at a random point, the RE exits at a consistent spot and yields back control.

So I'd then say it's of benefit to no one to even log the traceback, let alone print it to the user.